lib/axon/optimizers.ex

```diff
@@ -1,58 +1,5 @@
 defmodule Axon.Optimizers do
-  @moduledoc """
-  Implementations of common gradient-based optimization algorithms.
-
-  All of the methods in this module are written in terms of
-  the update methods defined in `Axon.Updates`. Axon treats
-  optimizers as the tuple:
-
-      {init_fn, update_fn}
-
-  where `init_fn` returns an initial optimizer state and `update_fn`
-  scales input gradients. `init_fn` accepts a model's parameters
-  and attaches state to each parameter. `update_fn` accepts
-  gradients, optimizer state, and current model parameters and
-  returns updated optimizer state and gradients.
-
-  Custom optimizers are often created via the `Axon.Updates` API.
-
-  ## Example
-
-  Consider the following usage of the Adam optimizer in a basic
-  update function (assuming `objective` and the `dataset` are
-  defined elsewhere):
-
-      defmodule Learning do
-
-        import Nx.Defn
-
-        defn init(params, init_fn) do
-          init_fn.(params)
-        end
-
-        defn update(params, optimizer_state, inputs, targets, update_fn) do
-          {loss, gradient} = value_and_grad(params, &objective(&1, inputs, targets))
-          {scaled_updates, new_optimizer_state} = update_fn.(gradient, optimizer_state, params)
-          {Axon.Updates.apply_updates(params, scaled_updates), new_optimizer_state, loss}
-        end
-      end
-
-      {model_params, _key} = Nx.Random.uniform(key, shape: {784, 10})
-      {init_fn, update_fn} = Axon.Optimizers.adam(0.005)
-
-      optimizer_state =
-        Learning.init(params, init_fn)
-
-      {new_params, new_optimizer_state, loss} =
-        Learning.update(params, optimizer_state, inputs, targets, update_fn)
-
-  For a simpler approach, you can also use optimizers with the training API:
-
-        model
-        |> Axon.Loop.trainer(:categorical_cross_entropy, Axon.Optimizers.adam(0.005))
-        |> Axon.Loop.run(data, epochs: 10, compiler: EXLA)
-
-  """
+  @moduledoc false
   alias Axon.Updates
```

Comment:
```suggestion
  alias Optimus.Updates
```

?

Reply:
I don't think we should do this. I believe `axon/{optimizers,updates,schedules}` should be left in this kind of "old version" kind of state until they are fully removed.

Reply:
I don't think we have an option, otherwise Axon will emit one bazilion warnings when compiling. Given the implementation is the same, I think we should do this change.

Reply:
Should we just fully delegate the whole modules then? Like with macro witchcraft

Reply:
We can do straight-forward delegation, I am not sure if it is worth macro-ing it, but I am also fine with leaving it as is. I would basically do the minimum amount of changes so this code is deprecated and compiles without warnings (as it is effectively dead code).